frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -154,12 +158,86 @@ private IGitRepositorySelectionProvider getGitRepositorySelectionProvider() {
 
   private void subscribeToGitRepositoryFilesChanges() {
     Topic<GitRepositoryChangeListener> topic = GitRepository.GIT_REPO_CHANGE;
-    GitRepositoryChangeListener listener = repository -> queueRepositoryUpdateAndModelRefresh();
+    GitRepositoryChangeListener listener = repository -> {
+      trackCurrentBranchChange(repository);
+      queueRepositoryUpdateAndModelRefresh();
+    };
+
     val messageBusConnection = project.getMessageBus().connect();
     messageBusConnection.subscribe(topic, listener);
     Disposer.register(project, messageBusConnection);
   }
 
+  private Notification getNotificationAboutNonManagedBranch(String branchName) {
+    val notification = VcsNotifier.STANDARD_NOTIFICATION.createNotification(
+        getString("action.GitMachete.EnhancedGraphTable.non-managed-branch-notification.text").format(branchName),
+        NotificationType.INFORMATION);
+    notification.addAction(
+        NotificationAction
```

Comment:
I'd also consider buttons like:
* _Don't suggest for this branch_
and/or
* _Don't suggest in this project_

Reply:
to avoid ppl getting annoyed 😅